src/gobble.py

```diff
@@ -60,19 +61,22 @@ def client_thread(routes_filter: Set[str]):
     url = f'https://api-v3.mbta.com/vehicles?filter[route]={",".join(routes_filter)}'
     logger.info(f"Connecting to {url}...")
     client = sseclient.SSEClient(requests.get(url, headers=HEADERS, stream=True))
-    current_stop_state: Dict = disk.read_state()
-    process_events(client, current_stop_state)
+    trips_state = TripsStateManager()
+    process_events(client, trips_state)
 
 
-def process_events(client: sseclient.SSEClient, current_stop_state: dict):
+def process_events(client: sseclient.SSEClient, trips_state: TripsStateManager):
     for event in client.events():
         try:
             if event.event != "update":
                 continue
             update = json.loads(event.data)
-            process_event(update, current_stop_state)
+            process_event(update, trips_state)
         except Exception:
-            logger.exception("Encountered an exception when processing an event", stack_info=True, exc_info=True)
+            if tracer.enabled:
```

Comment:
Any objections to my change here, which prints regular stack traces instead of JSON events locally? I find it much easier to work with — the JSON log events contain the traces, but without newlines, which is hard to read.

I could make this a config option or something. @devinmatte does this break a local Datadog workflow?

Reply:
Totally good. If you're using Datadog locally tracing would need to be enabled. This is a good change